Qt/QtMain.cpp

```diff
@@ -387,8 +402,27 @@ bool MainUI::event(QEvent *e)
     case QEvent::KeyRelease:
         NativeKey(KeyInput(DEVICE_ID_KEYBOARD, KeyMapRawQttoNative.find(((QKeyEvent*)e)->key())->second, KEY_UP));
         break;
+
     default:
-        return QWidget::event(e);
+		if (e->type() == browseFileEvent) {
+			QString fileName = QFileDialog::getOpenFileName(NULL, "Load ROM", g_Config.currentDirectory.c_str(), "PSP ROMs (*.iso *.cso *.pbp *.elf *.zip *.ppdmp)");
+			if (QFile::exists(fileName)) {
+				QDir newPath;
+				g_Config.currentDirectory = newPath.filePath(fileName).toStdString();
+				g_Config.Save();
+
+				NativeMessageReceived("boot", fileName.toStdString().c_str());
+			}
+			break;
+		} else if (e->type() == browseFolderEvent) {
+			I18NCategory *mm = GetI18NCategory("MainMenu");
+			QString fileName = QFileDialog::getExistingDirectory(nullptr, mm->T("Choose folder"), g_Config.currentDirectory.c_str());
+			if (QDir(fileName).exists()) {
+				NativeMessageReceived("browse_folderSelect", fileName.toStdString().c_str());
+			}
+		} else {
+	        return QWidget::event(e);
```

Comment:
Darn it, didn't even notice there were tabs and spaces in this file... sigh.

-[Unknown]